crypto/signals.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2016-2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "e_os.h"
+#include <string.h>
+
+#include <openssl/crypto.h>
+#include <internal/list.h>
+
+struct sigentry {
+    CRYPTO_SIGNAL p;
+    struct list list;
+};
+
+static struct list siglist;
```

Comment:
"struct list" is kind of generic, maybe "siglist"?  Also OpenSSL usually uses typedef'd struct's.

Reply:
It is used elsewhere in the code, notably in structs `crypto_worker_st` (crypto/thread/worker.h), `crypto_task_st` (crypto/thread/task.h), and `openssl_threads_st` (crypto/thread/openssl_threads.h). I am not sure renaming it to something more specific would fly. Or can you think of something less generic that would encompass the above as well as the above mentioned use in signals.c?

Reply:
sigthread?  a mixture of signal and thread?  So then you'd have
```
typedef struct sigthread_st {
. . .
} SIGTHREAD;
```

And use that typedef throughout.  Not a great name, admittedly, maybe someone on the project will come up with a better one.  The key point is using a typedef'd name for the struct.

Reply:
I realize that OpenSSL usually uses typedef'd structs. As `struct list` has only 2 pointers to struct list (prev and next) and carries no data, I thought that it would go a long way to suggest that it's internal. As for `SIGTHREAD` -- the list carries no data, so I am afraid that doesn't correspond well to reality and in some contexts the `SIG` prefix could be misleading.